docs/csharp/tour-of-csharp/tutorials/snippets/PatternMatching/Program.cs

```diff
@@ -0,0 +1,214 @@
+﻿// <InputData>
+string bankRecords = """
+    DEPOSIT,   10000, Initial balance
+    DEPOSIT,     500, regular deposit
+    WITHDRAWAL, 1000, rent
+    DEPOSIT,    2000, freelance payment
+    WITHDRAWAL,  300, groceries
+    DEPOSIT,     700, gift from friend
+    WITHDRAWAL,  150, utility bill
+    DEPOSIT,    1200, tax refund
+    WITHDRAWAL,  500, car maintenance
+    DEPOSIT,     400, cashback reward
+    WITHDRAWAL,  250, dining out
+    DEPOSIT,    3000, bonus payment
+    WITHDRAWAL,  800, loan repayment
+    DEPOSIT,     600, stock dividends
+    WITHDRAWAL,  100, subscription fee
+    DEPOSIT,    1500, side hustle income
+    WITHDRAWAL,  200, fuel expenses
+    DEPOSIT,     900, refund from store
+    WITHDRAWAL,  350, shopping
+    DEPOSIT,    2500, project milestone payment
+    WITHDRAWAL,  400, entertainment
+    """;
+// </InputData>
+
+IsTextValueExample(bankRecords);
+Console.WriteLine();
+IsEnumValueExample(bankRecords);
+Console.WriteLine();
+SwitchEnumValueExample(bankRecords);
+Console.WriteLine();
+ExampleProgram.Main();
+Console.WriteLine();
+
+static void IsTextValueExample(string bankRecords)
+{
+    // <IsOnTextValue>
+    double currentBalance = 0.0;
+    var reader = new StringReader(bankRecords);
+
+    string? line;
+    while ((line = reader.ReadLine()) != null)
```

Comment:
```suggestion
    while ((line = reader.ReadLine()) is not null)
```
To use even more pattern matching?